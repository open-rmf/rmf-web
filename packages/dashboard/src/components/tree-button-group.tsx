import { Button, ButtonGroup } from '@material-ui/core';
import ClearAllIcon from '@material-ui/icons/ClearAll';
import RestoreIcon from '@material-ui/icons/Restore';
import RestoreFromTrashIcon from '@material-ui/icons/RestoreFromTrash';
import React from 'react';

export interface TreeButtonGroupProps {
  disableReset?: boolean;
  disableClear?: boolean;
  disableRestore?: boolean;
  handleResetClick?: () => void;
  handleClearClick?: () => void;
  handleRestoreClick?: () => void;
  fullWidth?: boolean;
}

export const TreeButtonGroup = (props: TreeButtonGroupProps) => {
  const {
    disableReset,
    disableClear,
    disableRestore,
    handleResetClick,
    handleClearClick,
    handleRestoreClick,
    fullWidth = true,
  } = props;
  return (
    <div>
      <ButtonGroup fullWidth={fullWidth}>
        <Button
          id="reset-button"
          disabled={disableReset}
          onClick={() => handleResetClick && handleResetClick()}
        >
          <RestoreIcon />
          Reset
        </Button>
        <Button
          id="clear-button"
          disabled={disableClear}
          onClick={() => handleClearClick && handleClearClick()}
        >
          <ClearAllIcon />
          Clear
        </Button>
        <Button
          id="restore-button"
          disabled={disableRestore}
          onClick={() => handleRestoreClick && handleRestoreClick()}
        >
          <RestoreFromTrashIcon />
          Restore
        </Button>
      </ButtonGroup>
    </div>
  );
};
