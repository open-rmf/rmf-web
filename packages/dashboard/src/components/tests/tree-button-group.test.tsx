import { mount, shallow } from 'enzyme';
import React from 'react';
import { TreeButtonGroup } from '../tree-button-group';

test('Matches snapshot', () => {
  const root = shallow(<TreeButtonGroup />);
  expect(root).toMatchSnapshot();
});

test('It disabled buttons', () => {
  const root = mount(
    <TreeButtonGroup disableClear={true} disableReset={true} disableRestore={true} />,
  );
  expect(root.find('button#clear-button').props().disabled).toEqual(true);
  expect(root.find('button#reset-button').props().disabled).toEqual(true);
  expect(root.find('button#restore-button').props().disabled).toEqual(true);
});

test('It executes callbacks correctly', () => {
  const handleResetClick = jest.fn();
  const handleClearClick = jest.fn();
  const handleRestoreClick = jest.fn();
  const root = mount(
    <TreeButtonGroup
      handleResetClick={handleResetClick}
      handleClearClick={handleClearClick}
      handleRestoreClick={handleRestoreClick}
    />,
  );
  root.find('button#clear-button').simulate('click');
  root.find('button#reset-button').simulate('click');
  root.find('button#restore-button').simulate('click');

  expect(handleResetClick).toHaveBeenCalledTimes(1);
  expect(handleClearClick).toHaveBeenCalledTimes(1);
  expect(handleRestoreClick).toHaveBeenCalledTimes(1);
});
