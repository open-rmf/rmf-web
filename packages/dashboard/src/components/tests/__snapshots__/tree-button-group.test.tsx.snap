// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div>
  <WithStyles(ForwardRef(ButtonGroup))
    fullWidth={true}
  >
    <WithStyles(ForwardRef(Button))
      id="reset-button"
      onClick={[Function]}
    >
      <RestoreIcon />
      Reset
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button))
      id="clear-button"
      onClick={[Function]}
    >
      <ClearAllIcon />
      Clear
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button))
      id="restore-button"
      onClick={[Function]}
    >
      <RestoreFromTrashIcon />
      Restore
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(ButtonGroup))>
</div>
`;
